nixos/tests/ejabberd.nix

```diff
@@ -0,0 +1,303 @@
+import ./make-test.nix {
+  name = "ejabberd";
+  nodes = {
+    client = { nodes, pkgs, ... }: {
+      environment.systemPackages = let
+        sendMessage = pkgs.writeScriptBin "send-message" ''
+          #!/usr/bin/env python3
+          # Based on the sleekxmpp send_client example, look there for more details:
+          # https://github.com/fritzy/SleekXMPP/blob/develop/examples/send_client.py
+          import sleekxmpp
+
+          class SendMsgBot(sleekxmpp.ClientXMPP):
+              """
+              A basic SleekXMPP bot that will log in, send a message,
+              and then log out.
+              """
+              def __init__(self, jid, password, recipient, message):
+                  sleekxmpp.ClientXMPP.__init__(self, jid, password)
+
+                  self.recipient = recipient
+                  self.msg = message
+
+                  self.add_event_handler("session_start", self.start, threaded=True)
+
+              def start(self, event):
+                  self.send_presence()
+                  self.get_roster()
+
+                  self.send_message(mto=self.recipient,
+                                    mbody=self.msg,
+                                    mtype='chat')
+
+                  self.disconnect(wait=True)
+
+
+          if __name__ == '__main__':
+              xmpp = SendMsgBot("cthon98@example.com", "nothunter2", "azurediamond@example.com", "hey, if you type in your pw, it will show as stars")
+              xmpp.register_plugin('xep_0030') # Service Discovery
+              xmpp.register_plugin('xep_0199') # XMPP Ping
+
+              # TODO: verify certificate
+              # If you want to verify the SSL certificates offered by a server:
+              # xmpp.ca_certs = "path/to/ca/cert"
+
+              if xmpp.connect(('${nodes.server.config.networking.primaryIPAddress}', 5222)):
+                  xmpp.process(block=True)
+              else:
+                  print("Unable to connect.")
+                  sys.exit(1)
+        '';
+      in [ (pkgs.python3.withPackages (ps: [ ps.sleekxmpp ])) sendMessage ];
+    };
+    server = { config, pkgs, ... }: {
+      networking.extraHosts = ''
+        ${config.networking.primaryIPAddress} example.com
+      '';
+
+      services.ejabberd = {
+        enable = true;
+        configFile = "/etc/ejabberd.yml";
+      };
+
+      environment.etc."ejabberd.yml" = {
+        user = "ejabberd";
+        mode = "0600";
+        text = ''
+loglevel: 3
```

Comment:
This should be indented. The indentation will automatically be removed as explained in https://nixos.org/nix/manual/#idm140737317975776.